### DISCUSSION TITLE
Client / Server Code in Seperate Projects

### DISCUSSION THREAD

**Discussion by roestzwiee** [PlayFab/MpsSamples#47] [Q&A]
Hi All,

as a little exercise, I'm currently doing a small multiplayer game with Unity, Mirror, GSDK and PlayFab. I was happy when I found this example here, as it seemed to be the most up-to-date one. Now I managed to have the server running in PlayFab and being able to connect. Like in this example, I was happy to separate my client from my server. I got a little curious how generic my server would become as I started creating it, completely loose from my client.

Now my project of course contains of my own scenes what caused some trouble.
After a while of troubleshooting, I found that the NetworkManager docs state, that the server and the client need to have loaded the same scenes. 

How do you manage the separation between the client and the server scenes? Does it also make sense in non-educational / sample projects?

**Comment by dgkanatsios** [comment]
Hello and thanks for the feedback!
Most people use custom defines to separate between client and server Builds -> https://docs.unity3d.com/Manual/PlatformDependentCompilation.html

**Reply by roestzwiee** [reply]
Hello, thanks for your reply!

I get using platform dependent compilation. But this does not answer my question, as it is used to separate the code base. 

Is it correct to assume that in a real game scenario, one would **not** separate client and server **scenes** into different projects? 

The online scene in NetworkManager has to be the same for client and server. I can't find a reasonable way to separate it for client and server as e.g. Prefabs are referenced by generated IDs. Or am I missing something? 

-> Maybe with path references, one would be able to achieve a similar scene over different projects.

As it is done in this example, is it valid to have different offline scenes for server and client? To manage it in the same project, the NetworkManager could be extended with custom defines.
